awsagentprovider/src/main/java/software/amazon/opentelemetry/javaagent/providers/AwsMetricAttributeGenerator.java

```diff
@@ -457,6 +530,41 @@ private static void setRemoteResourceTypeAndIdentifier(SpanData span, Attributes
       builder.put(AWS_REMOTE_RESOURCE_TYPE, remoteResourceType.get());
       builder.put(AWS_REMOTE_RESOURCE_IDENTIFIER, remoteResourceIdentifier.get());
     }
+    if (cloudformationPrimaryIdentifier.isPresent()) {
+      builder.put(AWS_CLOUDFORMATION_PRIMARY_IDENTIFIER, cloudformationPrimaryIdentifier.get());
+    }
+  }
+
+  private static Optional<String> getLambdaNameFromArbitraryName(Optional<String> arbitraryName) {
+    AWSLambda lambdaClient = AWSLambdaClientBuilder.defaultClient();
+    GetFunctionRequest getFunctionRequest =
+        new GetFunctionRequest().withFunctionName(arbitraryName.get());
+    GetFunctionResult getFunctionResult = lambdaClient.getFunction(getFunctionRequest);
+    return Optional.of(getFunctionResult.getConfiguration().getFunctionName());
+  }
+
+  // NOTE: "name" in this case can be either the lambda name or lambda arn
+  private static Optional<String> getLambdaArnFromArbitraryName(Optional<String> arbitraryName) {
+    AWSLambda lambdaClient = AWSLambdaClientBuilder.defaultClient();
+    GetFunctionRequest getFunctionRequest =
+        new GetFunctionRequest().withFunctionName(arbitraryName.get());
+    GetFunctionResult getFunctionResult = lambdaClient.getFunction(getFunctionRequest);
+    return Optional.of(getFunctionResult.getConfiguration().getFunctionArn());
+  }
```

Comment:
I'm not sure if this is the best way to do this. 

According to the docs, the input for `getFunctionName()` can be the function name, function arn, or function alias.

- https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/lambda/model/GetFunctionRequest.html#getFunctionName--

Also, I wasn't sure if I should handle the alias case here but I can add that as well if necessary.

Reply:
I think it should be possible implement this functionality in upstream as well here: 

https://github.com/mxiamxia/opentelemetry-java-instrumentation/pull/4/files#diff-718d4039f7478ec457cfea44fa7c791e0e2a88f52302d782206511e990628d51

but I wasn't able to get it to work because getting the function arn is a multi-step process. 

`getFunction()` -> `GetFunctionResult object` -> `getConfiguration()` -> `FunctionConfiguration object` -> `getFunctionArn()`

https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/lambda/model/GetFunctionResult.html

Reply:
Yes. agree we should do it in upstream
Use the previous PR we implemented, will it work with modifications like 
```setAttribute(attributes, AWS_LAMBDA_FUNCTION_NAME, awsResps, requestAccess -> 
    Optional.ofNullable(requestAccess.getFunctionName())
            .map(functionName -> functionName.getConfiguration())
            .map(configuration -> configuration.getFunctionArn())
            .orElse(null)
);```
https://github.com/mxiamxia/opentelemetry-java-instrumentation/blob/c963b723720bf9008203357e78e32ad13578535a/instrumentation/aws-sdk/aws-sdk-1.11/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v1_11/AwsSdkExperimentalAttributesExtractor.java#L89

Reply:
Sounds good, I will try to implement this. Thanks for the suggestion!